inbm/dispatcher-agent/dispatcher/source/constants.py

```diff
@@ -25,22 +26,22 @@ class SourceParameters:
 
 @dataclass(kw_only=True, frozen=True)
 class ApplicationAddSourceParameters:
-    gpg_key_uri: str
-    gpg_key_name: str
-    file_name: str
-    sources: list[str] = field(default_factory=lambda: [])
+    source_list_file_name: str
+    sources: List[str] = field(default_factory=list)
```

Comment:
**Enhancement**
**Suggestion:** Since the ApplicationAddSourceParameters class is immutable, consider using a tuple instead of a list for the sources field to enforce immutability of the field's contents.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    @dataclass(kw_only=True, frozen=True)
    class ApplicationAddSourceParameters:
        sources: Tuple[str, ...] = field(default_factory=tuple)
```